### PR TITLE
Feature/copy-id

### PR DESCRIPTION
**Pull request by mixmix**
copying id wasn't working. 
Now works (minimally) in electron and firefox.

@pfraze would love some input on how i'm managing passing the parent to the child menu components so that I can modify state ... I tried refactoring what was there a little, but not sure what good style is yet.

Known problems - toggling raw view state doesn't seem to work reliably. Works more reliably than it used to, but not sure how to debug problem...

### CODE REVIEW

**Comment by pfraze** [comment]
I'll make some quick changes

**Comment by mixmix** [comment]
Thanks. Comments would be appreciated.

**Comment by mixmix** [comment]
If you have time, otherwise I'll check out commits

**Comment by pfraze** [comment]
Sure

**Comment by pfraze** [comment]
First update, the new per-item behavior for clicks was being used exclusively, and not falling back to the props.onSelect if unspecified.

**Comment by pfraze** [comment]
The data view failing to render had to do with an existing bug in `shouldComponentUpdate`.

**Comment by pfraze** [comment]
@mixmix why did you introduce the whole `faClass` thing?

**Comment by pfraze** [comment]
Oh, I think I see why. Ok, I prefer a different approach to that

**Comment by mixmix** [comment]
It was an attempt to dry up the code some, there was formerly a lot of repetition of structure. I extracted the unique parts. 
Interested to see your pattern

**Comment by pfraze** [comment]
In this case, I prefer to leave it as it was. Leaving label as a more general attribute bakes in fewer patterns and stays flexible.

**Comment by pfraze** [comment]
I didnt want the ClipboardBtn inside the dropdown, because that was inter-tangling a specific use-case with the dropdown, which should be more general. So, I added the `Com` attribute for providing a custom component on a dropdown option.

Unfortunately the ClipboardBtn also didnt call `stopPropagation` on the click event, so I had to rearrange the dom elements to keep clicking on the Copy ID btn from "reopening" the dropdown after it closed.

**Comment by pfraze** [comment]
Ok, I think this is ready to go now

**Comment by clehner** [comment]
@pfraze can I merge this into smarter-inbox-pt3?

**Comment by pfraze** [comment]
@clehner sure, though I figured it would be easier just to let it merge in when smarter-inbox-pt3 goes into master

**Comment by clehner** [comment]
Well, it's there now. Also the new unread state looks good.

**Comment by pfraze** [comment]
cool

File: ui/com/msg-view/card.jsx [hunk 1]
```diff
@@ -214,13 +227,24 @@ export default class Card extends React.Component {
...
-      { value: 'copy-link',  label: <span><i className="fa fa-external-link" /> Copy ID</span> },
-      { value: 'toggle-raw', label: <span><i className={isViewingRaw?'fa fa-envelope-o':'fa fa-gears'} /> View {isViewingRaw?'Msg':'Data'}</span> },
-      (isDownvoted) ?
-        { value: 'unflag',   label: <span><i className="fa fa-times" /> Unflag</span> } :
-        { value: 'flag',     label: <span><i className="fa fa-flag" /> Flag</span> }
+      {
+        value: 'copy-link',
+        Com: props => <ClipboardBtn component='li' data-clipboard-text={msg.key} onSuccess={props.onClick}>
+          <i className="fa fa-external-link" /> Copy ID
+        </ClipboardBtn>
```
Comment (mixmix) [inline ui/com/msg-view/card.jsx]: @pfraze this is pretty interesting. When this `item.Com` is used later, it doesn't get called like `item.Com(this.props)` 
Is that something that is implicitly done when jsx is transpiled  ?

i.e.
```jsx
<item.Com key={i} onClick={onClick} />

// becomes something like: 
<item.Com(this.props) key={i} onClick={onClick} />
```

tried googling stuff like _dynamic jsx component function_ ... not that much luck. If you've got keywords that would improve my search those would be great.

Comment (pfraze) [inline ui/com/msg-view/card.jsx]: Those are called stateless component functions, they're written about in this blog post https://facebook.github.io/react/blog/2015/09/10/react-v0.14-rc1.html

**Pull request merged**